shared/middleware/helpers/firebase_helper.rb

```diff
@@ -26,19 +26,48 @@ def table_as_csv(table_name)
     table_to_csv(records, column_order: ['id'])
   end
 
+  def number?(num)
+    !Float(num).nil? rescue false
+  end
+
+  def csv_as_table(csv_data)
+    records = {}
+    id = 1
+    table = CSV.parse(csv_data, headers: true)
+    table.each do |row|
+      record = OpenStruct.new
```

Comment:
why use an OpenStruct instead of a hash here? we're converting it to a hash on line 44